Enable 3D fields

Coolthulhu opened the pull request:
Currently only turns on fields outside current z-level, doesn't add any new spreading mechanics. Fires and gases will stay on their z-level of origin.

Also changed the descriptions of sounds. Instead of "center and above/below" it's just "above" or "below". It's still not perfect because the message says "you hear %s to the above", but it is a bit less awkward than it was.

Fire processing is really slow, I'll have to cut out some smoke processing before full 3D fields are doable.

BevapDin commented on src/line.cpp:
```diff
@@ -289,7 +289,8 @@ direction direction_from(int const x1, int const y1, int const x2, int const y2)
 direction direction_from(tripoint const &p, tripoint const &q)
 {
-    return direction_from(q.x - p.x, q.y - p.y, q.z - p.z);
+    // Note: Z coord has to be inverted either here or in direction defintions
```
I'd argue that this fix should go into the direction enum directly. The enum establishes a connection between the sign of the numbers and its observed meaning - a z-offset of -1 means down, just like a x-offset of -1 means west (left).

But this would change the enumerations numeric values, which is known to cause trouble (the older ones might remember the confused monster compass), so the fix here is probably good for now.

PropaneSoup commented:
This is starting to get exciting, we're pretty close to having this whole thing be 3D, aren't we?

BevapDin commented:
The sound symbol can easily overlay the symbol of creatures. Even through this is a debug/experimental feature, I'm not sure that's an acceptable behavior.

 Otherwise it works fine.

Coolthulhu commented:
Changed the sound display. Put it inside `draw_ter`, after terrain but before creature drawing.

Also added an ability to list sounds without description and sounds on different z-levels. Sounds without description will be described as "a sound", sounds on different z-levels will have a note saying they're from above/below.
